packages/extension/src/view/devtools/stateProviders/syncCookieStore/index.tsx

```diff
@@ -449,6 +457,306 @@ export const Provider = ({ children }: PropsWithChildren) => {
+            result = await chrome.cookies.set({
+              ...otherDetails,
+              expirationDate,
+              [keyToChange]: valueToBeSet,
+              url: cookieDetails.url,
+            });
+            if (keyToChange === 'expirationDate') {
+              valueToBeSet = Number(valueToBeSet) * 1000;
+            }
+          }
+
+          if (
+            result[keyToChange] === valueToBeSet ||
+            (keyToChange === 'expirationDate' &&
+              (result?.session || result?.expirationDate))
+          ) {
+            const oldData = await chrome.storage.local.get();
+
+            const updatedCookie = {
+              [newCookieKey]: {
+                ...cookieDetails,
+                parsedCookie: {
+                  ...cookieDetails.parsedCookie,
+                  [keyToChange === 'expirationDate'
+                    ? 'expires'
+                    : keyToChange.toLowerCase()]:
+                    getValueForLocalStorageAndCookieSet(
+                      keyToChange,
+                      valueToBeSet,
+                      true
+                    ),
+                },
+              },
+            };
+
+            await chrome.storage.local.set({
+              ...oldData,
+              [tabId]: {
+                cookies: {
+                  ...restOfCookies,
+                  ...updatedCookie,
+                },
+              },
+            });
+          } else {
+            return false;
+          }
+          return true;
+        } catch (error) {
+          return false;
+        }
+      } else {
+        return false;
+      }
+    },
+    [tabCookies, tabId]
+  );
+
+  const deleteAllCookies = useCallback(async () => {
+    const localStorage = await chrome.storage.local.get();
+    if (tabId && tabFrames && selectedFrame) {
+      const currentFrame = tabFrames[selectedFrame];
+      const newTabData = localStorage[tabId];
+      const allCookies: CookieStoreContext['state']['tabCookies'] =
+        newTabData.cookies ?? null;
+      if (allCookies) {
+        const cookieKeys: string[] = [];
+        await Promise.all(
+          currentFrame.frameIds.map(async (frameId) => {
+            await Promise.all(
+              Object.keys(allCookies).map(async (key) => {
+                if (allCookies[key].frameIdList.includes(frameId)) {
+                  if (
+                    allCookies[key]?.parsedCookie?.name &&
+                    allCookies[key]?.url
+                  ) {
+                    await chrome.cookies.remove({
+                      name: allCookies[key]?.parsedCookie?.name,
+                      url: allCookies[key]?.url,
+                    });
+                  }
+                  cookieKeys.push(key);
+                  return key;
+                }
+                return key;
+              })
+            );
+            return frameId;
+          })
+        );
+        await CookieStore.deleteSetOfCookie(cookieKeys);
+      }
```

Comment:
```suggestion
        if (!allCookies) {
          return;
        }
        
        const cookieKeys: string[] = [];
        await Promise.all(
          currentFrame.frameIds.map(async (frameId) => {
            await Promise.all(
              Object.keys(allCookies).map(async (key) => {
                  if (!allCookies[key].frameIdList.includes(frameId)) {
                    return key;
                  }
                
                  if (
                    allCookies[key]?.parsedCookie?.name &&
                    allCookies[key]?.url
                  ) {
                    await chrome.cookies.remove({
                      name: allCookies[key]?.parsedCookie?.name,
                      url: allCookies[key]?.url,
                    });
                  }
                  
                  cookieKeys.push(key);
                  return key;
              })
            );
            return frameId;
          })
        );
        await CookieStore.deleteSetOfCookie(cookieKeys);
```

Early return is helping in readability. Please verify the code before implementation.

Reply:
Fixed in latest commit.